src/orion/algo/space.py

```diff
@@ -64,7 +64,7 @@ class Dimension(object):
 
     """
 
-    NO_DEFAULT_VALUE = None
+    NO_DEFAULT_VALUE = object()
```

Comment:
Good one!

Reply:
But it does not work as-is unfortunately. Ultimately we should use something like this but we'll need to do it slightly differently. One of the reason if I remember correctly is that copy.deepcopy of the dimension fails since object() is not serializable.